src/Bicep.Core/Semantics/YamlObjectParser.cs

```diff
@@ -5,18 +5,35 @@
 using Bicep.Core.Text;
 using Newtonsoft.Json.Linq;
 using SharpYaml.Serialization;
+using System.IO;
 
 namespace Bicep.Core.Semantics
 {
     public class YamlObjectParser : ObjectParser
     {
+        private bool isMultiDocumentYaml = false;
+
         /// <summary>
         /// Deserialize raises an exception if the fileContent is not a valid YAML object
         /// </summary>
         override protected JToken? ExtractTokenFromObject(string fileContent)
         {
             try
             {
+                // Check for multi-document YAML using YamlStream
+                var yamlStream = new YamlStream();
+                using (var reader = new StringReader(fileContent))
+                {
+                    yamlStream.Load(reader);
+                }
+
+                if (yamlStream.Documents.Count > 1)
```

Comment:
Presumably if `Documents` count > 1 we _could_ return it an array of objects? If so, I'm wondering if we should just support multi-document yaml in `loadYamlContent` function?
cc @jeskew / @dciborow

Reply:
I don't see why not. `loadJsonContent` doesn't require that the target file be an object.

I'd say that it's kinda weird that transforming:

```yaml
abc: def
```

to

```yaml
abc: def
---
ghi: jkl
```

would change the output of loading the file from an object to an array, but that's just because yaml is weird.